std/src/main/scala/algebra/std/vector.scala

```diff
@@ -0,0 +1,41 @@
+package algebra
+package std
+
+import scala.annotation.tailrec
+
+package object vector extends VectorInstances
+
+trait VectorInstances {
+  implicit def vectorOrder[A: Order]: Order[Vector[A]] = new VectorOrder[A]
+  implicit def vectorMonoid[A]: Monoid[Vector[A]] = new VectorMonoid[A]
+}
+
+final class VectorOrder[A](implicit ev: Order[A]) extends Order[Vector[A]] {
+  def compare(x: Vector[A], y: Vector[A]): Int = {
+    val lengthX = x.length
+    val lengthY = y.length
+
+    @tailrec
+    def cmp(index: Int): Int = {
+      if (index == lengthX) {
+        lengthX - lengthY
+      } else if (index == lengthY) {
+        1
+      } else {
+        val n = ev.compare(x(index), y(index))
+        if (n == 0) cmp(index + 1) else n
+      }
+    }
+
+    cmp(0)
+  }
+}
+
+class VectorMonoid[A] extends Monoid[Vector[A]] {
```

Comment:
Agreed. Also it would be a good idea to figure out if we want to endorse this as the *the* `Monoid[Vector[A]]` instance (since it is parametric) or merely as an alternative to `Monoid[A] => Monoid[Vector[A]]` (since that is what folks treating vectors as vectors probably want).